### PR TITLE
Enhance error handling in clients

### PR DESCRIPTION
**Pull request by 0oM4R**
### Changes

Add two types to types package: 
- `TwinDoesNotExistError` 
- `InsufficientBalanceError`

when user logged in and the balance is lower than 0.01 TFT, a toast will notify the user to fund the account 
 
image

[Screencast from 21 ديس, 2023 EET 06:24:19 م.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/9da60280-79bb-47aa-abf0-4c12f391c540)

enhance the order of connecting to the clients to handle errors properly 

### Related Issues

- #1665
- #1597 

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/rmb_direct_client/lib/client.ts [hunk 1]
```diff
@@ -100,25 +109,19 @@ class Client {
...
       const twinId = await this.tfclient.twins.getTwinIdByAccountId({ accountId: this.signer.address });
-      this.twin = await this.tfclient.twins.get({ id: twinId });
       if (!twinId) {
-        throw new ValidationError(`Couldn't find a user for the provided mnemonic on this network.`);
+        throw new TwinDoesNotExistError(`Couldn't find a user for the provided mnemonic on this network.`);
       }
 
+      this.twin = await this.tfclient.twins.get({ id: twinId });
       try {
         if (!this.twin) {
```
Comment (AhmedHanafy725) [inline packages/rmb_direct_client/lib/client.ts:123]: as you were asking @0oM4R,  this is a repeated check. you can remove it

File: packages/grid_client/src/client.ts [hunk 2]
```diff
@@ -121,6 +126,7 @@ class GridClient {
       process.on("SIGINT", this.disconnectAndExit);
       process.on("SIGUSR1", this.disconnectAndExit);
       process.on("SIGUSR2", this.disconnectAndExit);
+      //TODO move this line to disconnectAndExit
```
Comment (amiraabouhadid) [inline packages/grid_client/src/client.ts:129]: why is this a todo and not done yet?

**Review by amiraabouhadid — COMMENTED** [review]
looks good
image

File: packages/grid_client/src/client.ts (hunk 2 above)
Comment (0oM4R) [reply packages/grid_client/src/client.ts:129]: Will resolve it later in another PR

File: packages/tfchain_client/src/client.ts [hunk 3]
```diff
@@ -351,15 +365,19 @@ class Client extends QueryClient {
           resultSections.length > 0
         )
           section = resultSections[0];
+        //TODO no need to do the following lines only check if the error is instanceof base error and edit the message
```
Comment (maayarosama) [inline packages/tfchain_client/src/client.ts:368]: Is this to be worked on in another pr? if so please open an issue and mention it here

File: packages/grid_client/src/client.ts (hunk 2 above)
Comment (maayarosama) [reply packages/grid_client/src/client.ts:129]: Open and issue and mention it here too

File: packages/tfchain_client/src/client.ts (hunk 3 above)
Comment (0oM4R) [reply packages/tfchain_client/src/client.ts]: removed

File: packages/grid_client/src/client.ts (hunk 2 above)
Comment (0oM4R) [reply packages/grid_client/src/client.ts:129]: https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1781

**Review by maayarosama — APPROVED** [review]

**Review by amiraabouhadid — APPROVED** [review]
